client/src/app.jsx

```diff
@@ -1,57 +1,103 @@
+  //     const counter = current[1];
+  //     reviewCount += counter;
+  //     reviewTotal += stars * counter;
+  //   }
+  //   console.log('App: logging total and count', reviewTotal, reviewCount);
+  //   return reviewTotal / reviewCount;
+  // };
+  // const getTheReviewCount = (starData) => {
+  //   let reviewCount = 0;
+  //   const ratingArr = Object.entries(starData);
+  //   for (let i = 0; i < ratingArr.length; i += 1) {
+  //     const current = ratingArr[i];
+  //     const counter = current[1];
+  //     reviewCount += counter;
   //   }
-  // })
+  //   return reviewCount;
+  // };
+
+  useEffect(() => {
+    getAllStyles((error, response) => {
+      if (error) {
+        return 'Could not get styles';
+      }
+      return setStyles(response);
+    });
+    getInfo((error, response) => {
+      if (error) {
+        return 'Could not get products';
+      }
+      setId(response.id);
+      return setProduct(response);
+    });
+    getReviews((error, response) => {
+      if (error) {
+        return 'Could not get reviews';
+      }
+      console.log('App: get reviews response', response.ratings);
+      return setReviews(response.ratings); // object
+    });
+    // getRating((error, response) => {
+    //   if (error) {
+    //     return 'Could not get ratings';
+    //   }
+    //   console.log('App: response from getRating', response);
+    //   const average = getTheAverageRating(response);
+    //   return getRating(average);
+    // });
+    // getCount((error, response) => {
+    //   if (error) {
+    //     return 'Could not get review count';
+    //   }
+    //   console.log('App: response from getCount', response);
+    //   const numberOReviews = getTheReviewCount(response);
+    //   return getCount(numberOReviews);
+    // });
+  }, []);
 
-  render() {
+  // console.log('App: rating before passing to product info: ', rating);
+  // console.log('App: count before passing to product info: ', count);
+  if (Object.keys(reviews).length === 0
+  || Object.keys(product).length === 0
+  || Object.keys(styles).length === 0) {
     return (
-      <Container>
-        <br />
-        <TitleImage />
-        <Container>
-          <Row>
-            <MainImage />
-            Stars will go here
-            <br />
-            Category will go here
-            <br />
-            Product Name will go here
-            <br />
-            Price will go here
-            <br />
-            Style and Selected Style will go here
-            <br />
-            Style Thumbnails will go here
-            <br />
-            Size and Quantity Dropdowns will go here
-            <br />
-            Add to Cart and Favorite Buttons will go here
-            <br />
-          </Row>
-        </Container>
-        <Container>
-          Product Description will go here
-        </Container>
-      </Container>
+      <div />
     );
   }
-}
+  console.log('reviews before passing to product info: ', reviews);
```

Comment:
Console log here could be misleading as the state will not line up with it on the first run through. I could be wrong here too

Reply:
Because it comes after the if state is undefined don't render anything it has been working. I no longer need this log luckily!